src/extension.ts

```diff
@@ -232,11 +238,100 @@ function activateLsp(context: ExtensionContext) {
   client.start()
 }
 
+async function restartOnEdits(context: ExtensionContext) {
+  // TODO: handle case where sgconfig.yml is not in root
+  // TODO: handle case where configPath is not set to sgconfig.yml
+  // TODO: handle case where no workspace folder is open and then we open one
+  // TODO: make sure dispose actually does what it should do
+  // TODO: handle more than 1 element in ruleDirs
+
+  if (!workspace.workspaceFolders || !workspace.workspaceFolders[0]) {
+    return
+  }
+
+  let ruleWatcher: FileSystemWatcher | undefined
+  let disposables: Disposable[] = []
+  async function getRuleDirs(configPath: string): Promise<string[]> {
```

Comment:
Parsing yaml in JS looks like sub-optimal since it is duplicating server works. 
Maybe we can have a special command to server to report watched files.
That said, I think the current approach is good enough at the moment. We can improve it later